text/dependencies/rfcs/0000-dependency-management-phase-one.md

```diff
@@ -0,0 +1,300 @@
+| pipenv            | https://files.pythonhosted.org/packages                   | yes, with other pip deps buildpacks-ci   |
+| python            | https://www.python.org/ftp/python                         | yes, buildpacks-ci                       |
+| poetry            | https://files.pythonhosted.org/packages                   | processed, buildpacks-ci                 |
+| ruby              | https://cache.ruby-lang.org/pub/ruby                      | yes, binary-builder                      |
+| rust              | https://static.rust-lang.org/dist/                        | yes, buildpacks-ci                       |
+| tini              | https://github.com/krallin/tini/tarball                   | yes, buildpacks-ci                       |
+| yarn              | https://github.com/yarnpkg/yarn/releases/download/v1.15.2 | processed, buildpacks-ci                 |
+</details>
+
+### Transfer Compilation Code
+
+For any dependencies that must still be compiled or processed in some way, the
+code from the Cloud Foundry repositories should be rewritten,
+and moved over to the buildpack under a directory called
+`dependency/compilation`. The code should be easily runnable from a local
+machine to compile the code, and should be documented with a README about how
+to use it. The dependency will likely need to be compiled against different
+stacks/architectures, so this should also be enumerated in the RFC and taken
+account of in the new compilation code. It will have the metadata for a
+dependency available as an input for determining source URI, version, and
+compatible stacks.
+
+Maintainers should also consider if the dependency of interest if OS
+distribution-agnostic, or if it will need to be compiled separately depending
+on the distribution or platform it's used on.
+
+The code will eventually be used in a Github Actions workflow, so the location
+needs to be standardized across buildpacks. The workflows and Github Actions to
+use this code will be enumerated in a separate RFC.
+
+### Bucket Setup
+
+The new dependency management automation (described in a subsequent RFC) will
+rely on the `buildpack.toml` and dependency-specific code as the source of
+truth for the latest metadata and known versions. This will lead to greater
+transparency and discoverability in how we update dependencies.
+
+Since metadata and known versions won't be stored data anymore, buckets will
+only be needed to store dependencies that have been compiled or processed in
+some way. For these cases, a dependency-specific bucket set up in Paketo
+Dependencies project within GCP to store data. The directory name will be the
+name of the dependency and it will contain dependency archives (if compiled).
+Push credentials to this bucket will be reserved for the buildpack and
+maintainers, but the contents will be publicly available.
+
+### Remove Known Versions
```

Comment:
So my understanding of having known versions is so that not only can the buildpack pick from a set of versions but so the retrieval logic knows what buildpacks it has already seen before. Is the main assumption going forward going to be that if the version number is greater than the one of the lines we are checking for it is new? Just a thought and would like to hear what the general expected strategy is.

Reply:
Ok I am reading into phase two now and that does seem to answer this question. A quick shout out here would be appreciated.

Reply:
you got it. Done in https://github.com/paketo-buildpacks/rfcs/pull/205/commits/cb117e4f45d2bc4e8d7551667100b5c8b9d7e89b